### PR TITLE
Task3 added.

### PR DESCRIPTION
**Pull request by teadrunkage**
Daria Schek., 626

### CODE REVIEW

**Comment by nefanov** [comment]
Прокомментировал.

**Comment by teadrunkage** [comment]
Task2 not yet continuous, working on it.

File: task_1/task/1_cmd_runner.c [hunk 1]
```diff
@@ -16,22 +16,58 @@
...
+			}
+			else {
+				printf("See you.\n");
+				q = 1;
+			//	break;
+			}
+			
+		} else {
+			char * str = (char *) malloc (255 * sizeof(char));
+			char * * args = (char * *) malloc (4096 * sizeof(char *));
```
Comment (nefanov) [inline task_1/task/1_cmd_runner.c]: Принял, но строки коротковаты:
"
<<echo Hello
>>Hello
Ret code: 0
<<echo ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDkDpQGHj4SsU3omRu2t1SaCLq+SG/Z2MnvrWvRI59UCQLy7LwZWLX1ULODq045Gp+0c2Ow35A7lfc+kxMTnQgyCFq40mi4KUHsdA40bS0N+LfoOK5fCFNjQczxcPwzbUxiriG7UQhMGG41HFxsBemu6Druxpjw52FFzqsPersVcQg0Jm4cBZ7lmzDX0U7PZdeKYNv9JEXMz4P2mGlCJ9HDsBFU2BfYnLBoyBhsJ9lxnI+BxXY4fug9peo5t2z+XZgerBCYwNsmvg1B9FJKDzF6WU5sg1bF/TedvRAPkSHfXgooqKf9iZaKJJgB3YEoRL7EDzqzsJ0Bz6yPJC5pt5Pp
>>Error on start
Ret code: 42"

File: task_2/task.c [hunk 2]
```diff
@@ -8,3 +8,98 @@ typedef struct {
...
+		
+		close(mypipe.txd[0]);
+		printf("What will the parent say to the child?\n");
+		gets(str);
+		size = write(mypipe.txd[1], str, sizeof(str));
+		if (size != sizeof(str)) {
+			printf("Can't wright the string\n");
+			exit(-1);
+		}
+		close(mypipe.txd[1]);
```
Comment (nefanov) [inline task_2/task.c]: Один единственный вызов и закрытие дескриптора? А гарантировать целостность ?

**Comment by teadrunkage** [comment]
Зависимость времени от числа потоков выглядит странно

n = 1:
real	0m0.058s
user	0m0.052s
sys	0m0.000s

n = 2:
real	0m0.648s
user	0m0.496s
sys	0m0.784s

n = 3:
real	0m0.181s
user	0m0.200s
sys	0m0.280s

n = 4:
real	0m0.407s
user	0m0.464s
sys	0m1.000s

n = 5:
real	0m0.387s
user	0m0.448s
sys	0m0.964s

n = 6:
real	0m0.375s
user	0m0.400s
sys	0m0.980s

n = 7:
real	0m0.367s
user	0m0.292s
sys	0m1.116s

n = 8:
real	0m0.359s
user	0m0.356s
sys	0m1.028s

n = 9:
real	0m0.377s
user	0m0.372s
sys	0m1.080s

n = 10:
real	0m0.364s
user	0m0.392s
sys	0m1.016s

**Pull request closed without merging**